marketplacetests/marketplace/app.py

```diff
@@ -65,14 +64,18 @@ def offline_message_text(self):
         self.wait_for_element_displayed(*self._offline_message_locator)
         return self.marionette.find_element(*self._offline_message_locator).text
 
-    def wait_for_notification_message_displayed(self, message=None):
-        if message:
-            Wait(marionette=self.marionette).until(
-                lambda m: self.notification_message == message)
-        self.wait_for_element_displayed(*self._notification_locator)
+    def wait_for_notification_message(self, message):
+        element = self.marionette.find_element(*self._notification_locator)
+        # Wait for the expected message
+        Wait(marionette=self.marionette).until(lambda m: element.text == message)
+        # Wait for the message to no longer be displayed
+        Wait(marionette=self.marionette).until(expected.element_not_displayed(element))
```

Comment:
Waiting for an element to be not displayed without ever verifying that it is displayed always triggers an alarm for me. Are you confident this is safe? Can we first wait for the notification to be displayed?